docker/docker-compose.yml

```diff
@@ -0,0 +1,83 @@
+# U.S. Government sponsorship acknowledged.
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# • Redistributions of source code must retain the above copyright notice,
+#   this list of conditions and the following disclaimer.
+# • Redistributions must reproduce the above copyright notice, this list of
+#   conditions and the following disclaimer in the documentation and/or other
+#   materials provided with the distribution.
+# • Neither the name of Caltech nor its operating division, the Jet Propulsion
+#   Laboratory, nor the names of its contributors may be used to endorse or
+#   promote products derived from this software without specific prior written
+#   permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+version: "3.9"
+
+services:
+  # Registry loader component with Harvest and Registry Manager
+  registry-loader:
+    profiles: ["reg-loader"]
+    image: ${REG_LOADER_IMAGE}
+    environment:
+      - ES_URL=${ES_URL}
+    volumes:
+      - ${HARVEST_CFG_FILE}:/cfg/harvest-config.xml
+      - ${HARVEST_DATA_DIR}:/data
+    networks:
+      - pds
+
+    # Registry loader component with Harvest and Registry Manager
+  registry-loader-test:
+    profiles: [ "reg-loader-test" ]
+    image: ${REG_LOADER_IMAGE}
+    environment:
+      - ES_URL=${ES_URL}
+      - RUN_TESTS=true
+      - TEST_DATA_URL=${TEST_DATA_URL}
+    networks:
+      - pds
+
+  # Elasticsearch
+  elasticsearch:
+    profiles: ["elastic","services"]
+    image: ${ES_IMAGE}
+    environment:
+      - discovery.type=${ES_DISCOVERY_TYPE}
+    ports:
+      - "9200:9200"
+      - "9300:9300"
+    networks:
+      - pds
+
+  # Registry API
+  registry-api:
+    profiles: ["api","services"]
+    image: ${REG_API_IMAGE}
+    ports:
+      - "8080:8080"
+    volumes:
+      - ${REG_API_APP_PROPERTIES_FILE}:/usr/local/registry-api-service/application.properties
+      - ${REG_API_WAIT_FOR_ES_SCRIPT}:/usr/local/registry-api-service/wait-for-elasticsearch.sh
+    networks:
+      - pds
+    command: /usr/local/registry-api-service/wait-for-elasticsearch.sh
+
+networks:
+  pds:
```

Comment:
This is more a matter of my own personal taste, but I'm not tremendously fond of seeing a bare key in a yaml with no values. I get that we just want the defaults, but sometimes being explicit about those defaults is nice. And you can also apply metadata here so people who are debugging a system later can see what's going on:
```yaml
networks:
    pds:
        driver: bridge
        labels:
            org.label-schema.name: Planetary Data System (PDS)
            org.label-schema.description: Internal bridge network for interprocess communication of services for the PDS
```

Reply:
Very valid point! Updated the code as suggested!